### PR TITLE
Adopt PR template

### PR DESCRIPTION
**Pull request by jurre**
### What are you trying to accomplish?

In order to ensure that PRs are easy to review and of high quality, let's adopt this simple PR template similar to what we use internally.

### Anything you want to highlight for special attention from reviewers?

Any items we should add or remove to the checklist? Anything else you'd like to see reworded, added or removed?

### How will you know you've accomplished your goal?

New PRs include the details that are requested by this template.

### Checklist

- [x] I have run the complete test suite to ensure all tests and linters pass.
- [x] I have thoroughly tested my code changes to ensure they work as expected, including adding additional tests for new functionality.
- [x] I have written clear and descriptive commit messages.
- [x] I have provided a detailed description of the changes in the pull request, including the problem it addresses, how it fixes the problem, and any relevant details about the implementation.
- [x] I have ensured that the code is well-documented and easy to understand.

### CODE REVIEW

**Comment by robaiken** [comment]
should we request an issue before an PR is created?

**Review by thavaahariharangit — APPROVED** [review]
LGTM :)

**Comment by jurre** [comment]
> should we request an issue before an PR is created?

I would say for larger changes yes, for small bugfixes it might be too much overhead

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+### Checklist
+
+<!-- Before requesting review, please ensure that your pull request fulfills the following requirements: -->
+
+- [ ] I have run the complete test suite to ensure all tests and linters pass.
+- [ ] I have thoroughly tested my code changes to ensure they work as expected, including adding additional tests for new functionality.
+- [ ] I have written clear and descriptive commit messages.
+- [ ] I have provided a detailed description of the changes in the pull request, including the problem it addresses, how it fixes the problem, and any relevant details about the implementation.
+- [ ] I have ensured that the code is well-documented and easy to understand.
```
Comment (landongrindheim) [inline .github/pull_request_template.md:27]: 🤔 If we include these expectations in the pull request template, I think it makes sense to ensure they're in CONTRIBUTING.md. As far as I can tell, we're mostly covered 👇 

- ✅ Tests passing [is implied](https://github.com/dependabot/dependabot-core/blob/26239220c4a230356d687ac2ebd0d47195a0573f/CONTRIBUTING.md?plain=1#L19-L20)
- ✅ Adding tests is [explicitly listed](https://github.com/dependabot/dependabot-core/blob/26239220c4a230356d687ac2ebd0d47195a0573f/CONTRIBUTING.md?plain=1#L19)
- ✅ Commit messages are [well covered](https://github.com/dependabot/dependabot-core/blob/26239220c4a230356d687ac2ebd0d47195a0573f/CONTRIBUTING.md?plain=1#L27-L33)
- ✅ Description of the changes is in this template 🎉
- ⬜︎ I don't see anything about documentation or ease of understanding

I think it'd be fine to add this expectation in a followup PR, but IMO we should state this clearly in CONTRIBUTING.md if we expect it from contributors. What do you think?

**Review by landongrindheim — APPROVED** [review]

File: .github/pull_request_template.md (hunk 1 above)
Comment (jurre) [reply .github/pull_request_template.md:27]: Good shout, done!

**Review by landongrindheim — APPROVED** [review]
Love it! 📈

**Review by landongrindheim — APPROVED** [review]

**Pull request merged**